Plutarch.hs

```diff
@@ -19,7 +19,6 @@ module Plutarch (
   PI.plet,
   PI.punsafeBuiltin,
   PI.punsafeCoerce,
-  PI.punsafeConstant,
```

Comment:
I decided to unexport this to create an expectation that users should not use it, and instead should use `pconstant`.

Reply:
I get this, but generally I think we should preserve backward compatibility. It's better to just deprecate it IMO.

Reply:
I think that's generally a good idea -- and I'll add deprecation notice to `pdataLiteral` and friends -- but we can't deprecate `punsafeConstant` yet, because it is used internally by Plutarch:

<img width="691" alt="image" src="https://user-images.githubusercontent.com/3998/148132118-69fdc60d-7173-4497-945f-1fd44aadb6a9.png">

Reply:
By the way, in this PR `punsafeConstant` is used only in two places:
- `Plutarch.Lift`
- `uplcTests` that you added.

Reply:
We can `-Wno-deprecate` the relevant modules, but that's quite ugly. A better solution is to create a new name for `punsafeConstant` (e.g. `punsafeConstantInternal`), make `punsafeConstant` be an alias for that, and only deprecate the alias.

Reply:
Done.